working/augmentation-libraries/feature-specification.md

```diff
@@ -85,58 +85,57 @@ imports and prefixes of their parent (part or library) file.
+in the file itself, or if any of the files included by the Dart file contain
+the declaration. _That is, if the declaration occurs in a file in the subtree
 of that Dart file._
 
-We then define a partial and a complete *ordering* of declarations of a library
-as follows:
+We then define two *orderings* of declarations in a library, one partial and one
+complete, as follows:
 
 We define a partial ordering on syntactic declarations of a library,
 *is above*, such that a syntactic declaration *A* is *above* a syntactic
 declaration *B* if and only if:
 
-*   *A* and *B* occur in the same file, and the start of the *A* declaration is
-syntactically before the start of the *B* declaration, in source order, or
-*   A file included by the file containing *A* contains *B*.
+*   *A* and *B* occur in the same file, and the start of *A* is syntactically
+    before the start of *B*, in source order, or
+*   The file containing *A* includes the file that contains *B*.
 
 We define a *total ordering relation* (transitive, anti-symmetric, irreflexive)
 on declarations of a library, *is before* (and its reverse, *is after*) such
 that for any two syntactic declarations *A*, and *B*:
 
 *   If *A* and *B* occur in the same file, then:
-    *   If the start of *A* is before the start of *B* in source order,
-        then *A* is before *B*.
+    *   If the start of *A* is syntactically before the start of *B* in source
+        order, then *A* is before *B*.
+    *   Otherwise *B* is before *A*.
+*   Otherwise *A* and *B* occur in different files:
+    *   Let *F* be the least containing file for those two files.
+    *   If *A* occurs in *F* then *A* is before *B*.
+    *   If *B* occurs in *F* then *B* is before *A*.
+    *   Otherwise *A* and *B* are contained in distinct included files of *F*.
+    *   If the `part` directive in *F* including the file that contains *A* is
+        syntactically before the `part` directive in *F* including the file that
+        contains *B* in source order, then *A* is before *B*.
     *   Otherwise *B* is before *A*.
-*   Otherwise *A* and *B* occur in different files.
-*   Let *F* be the least containing file for those two files.
-*   If *A* occurs in *F* then *A* is before *B*.
-*   If *B* occurs in *F* then *B* is before *A*.
-*   Otherwise *A* and *B* are contained in distinct included files of *F*.
-*   If the `part` directive including the file that contains *A*
-    is syntactically before the `part` directive including the file that
-    contains *B* in source order, then *A* is before *B*.
-*   Otherwise *B* is before *A*.
 
 Then *B* *is after* *A* if and only if *A* *is before* *B*.
 
-(Here the first five points can be summarized as “If *A* is above *B*, then *A*
-is before *B*, and vice versa” and the remaining case covers when the two are
-contained in sibling part directives, and at least one of those three cases
-must occur.)
+*In short, if *A* is above *B*, then *A* is before *B*, and vice versa.
```

Comment:
'vice versa' is dangerous here, we don't want to imply "if *A* is before *B*, then *A* is above *B*".

Of course, we could say that "*above* implies *before*", but this puts the focus on the relations rather than individual elements, which might be considered _less_ readable. ;-)

On the other hand, 'vice versa', in the sense that makes sense, just means "if *B* is above *A*, then *B* is before *A*". But this doesn't bring anything new, because it's just a different assignment of a concrete example to a couple of metavariables.

So I'd suggest that we simply omit 'vice versa'.

Reply:
ACK. Better without.

Reply:
Good call. Done.